microsoft/testsuites/core/provisioning.py

```
@@ -268,6 +268,37 @@ def verify_stop_start_in_platform(
             is_restart=False,
         )
 
+    @TestCaseMetadata(
+        description="""
+        This case performs a reboot stress test on the node
+        and iterates smoke test 100 times.
+        The test steps are almost the same as `smoke_test`.
+        The reboot times is summarized after the test is run
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+            supported_features=[SerialConsole],
+        ),
+    )
+    def stress_reboot(self, log: Logger, node: RemoteNode, log_path: Path) -> None:
+        reboot_times = []
+        try:
+            for i in range(100):
+                log.info(f"Reboot stress iteration {i+1}/100")
+                elapsed = self._smoke_test(log, node, log_path, "stress_reboot")
+                reboot_times.append((i + 1, elapsed))
+
+        finally:
+            log.info(f"Number of iterations completed: {len(reboot_times)}")
+            log.info("Reboot times for all iterations:")
+            for iteration, time in reboot_times:
+                log.info(f"Iteration {iteration}: Reboot time = {time}s")
+            if len(reboot_times) < 100:
```

Use assert that to assert the number. The error message is clearer to show the actual and expected values.

---

BTW, I don’t think we need to assert the number here. If the reboot fails, an exception should already be thrown. Adding an assertion just rethrows the error and could obscure the original message.

---

Incase of a passedexception , even when the number of iterations is less than 100, the test passes with a warning

---

I see. But how would you like to deal with the PassedException in this case? Consider it's failed or passed? The assertError consider it's failed. If so, you can just rethrow the exception to other types, if it's passed exception. The overall logic is like below,

```python
try:
  # current try part
except PassedException as e:
  raise LISAException(e.message)
finally:
  # current finally part
```